module load gcc/4.8.2-fasrc01 ngsTools/20170201-fasrc01

angsd -bam /n/holylfs/LABS/edwards_lab/ashultz/HFWGReseq/whole-genome-reseq/BamLists_NoRel /CA_01_BamList.txt -anc /n/holylfs/LABS/edwards_lab/ashultz/HFWGReseq/gatk/anc_fasta/HF_ancestral_alleles.fasta -uniqueOnly 1 -remove_bads 1 -trim 0 -minMapQ 20 -minQ 20 -sites /n/holylfs/LABS/edwards_lab/ashultz/HFWGReseq/genome/bedfiles/All_HF_Whole_Genome_Include_Autosomal_Sites_d50.angsdsites -only_proper_pairs 1 -nThreads 8 -out results_saf_subpops/CA_01_Int10 -doSaf 2 -GL 1 -skipTriallelic 1 -rf split_fai_20_angsd/final.assembly.homo.fa_10.interval_list

angsd -bam /n/holylfs/LABS/edwards_lab/ashultz/HFWGReseq/whole-genome-reseq/BamLists_NoRel /CA_01_BamList.txt -anc /n/holylfs/LABS/edwards_lab/ashultz/HFWGReseq/gatk/anc_fasta/HF_ancestral_alleles.fasta -uniqueOnly 1 -remove_bads 1 -trim 0 -minMapQ 20 -minQ 20 -sites /n/holylfs/LABS/edwards_lab/ashultz/HFWGReseq/genome/bedfiles/All_HF_Whole_Genome_Include_Autosomal_Sites_d50.angsdsites -only_proper_pairs 1 -nThreads 8 -out results_saf_subpops/CA_01_Int8 -doSaf 2 -GL 1 -skipTriallelic 1 -rf split_fai_20_angsd/final.assembly.homo.fa_8.interval_list

angsd -bam /n/holylfs/LABS/edwards_lab/ashultz/HFWGReseq/whole-genome-reseq/BamLists_NoRel /IL_14_BamList.txt -anc /n/holylfs/LABS/edwards_lab/ashultz/HFWGReseq/gatk/anc_fasta/HF_ancestral_alleles.fasta -uniqueOnly 1 -remove_bads 1 -trim 0 -minMapQ 20 -minQ 20 -sites /n/holylfs/LABS/edwards_lab/ashultz/HFWGReseq/genome/bedfiles/All_HF_Whole_Genome_Include_Autosomal_Sites_d50.angsdsites -only_proper_pairs 1 -nThreads 8 -out results_saf_subpops/IL_14_Int17 -doSaf 2 -GL 1 -skipTriallelic 1 -rf split_fai_20_angsd/final.assembly.homo.fa_17.interval_list

angsd -bam /n/holylfs/LABS/edwards_lab/ashultz/HFWGReseq/whole-genome-reseq/BamLists_NoRel /MA_14_BamList.txt -anc /n/holylfs/LABS/edwards_lab/ashultz/HFWGReseq/gatk/anc_fasta/HF_ancestral_alleles.fasta -uniqueOnly 1 -remove_bads 1 -trim 0 -minMapQ 20 -minQ 20 -sites /n/holylfs/LABS/edwards_lab/ashultz/HFWGReseq/genome/bedfiles/All_HF_Whole_Genome_Include_Autosomal_Sites_d50.angsdsites -only_proper_pairs 1 -nThreads 8 -out results_saf_subpops/MA_14_Int4 -doSaf 2 -GL 1 -skipTriallelic 1 -rf split_fai_20_angsd/final.assembly.homo.fa_4.interval_list

angsd -bam /n/holylfs/LABS/edwards_lab/ashultz/HFWGReseq/whole-genome-reseq/BamLists_NoRel /ME_01_BamList.txt -anc /n/holylfs/LABS/edwards_lab/ashultz/HFWGReseq/gatk/anc_fasta/HF_ancestral_alleles.fasta -uniqueOnly 1 -remove_bads 1 -trim 0 -minMapQ 20 -minQ 20 -sites /n/holylfs/LABS/edwards_lab/ashultz/HFWGReseq/genome/bedfiles/All_HF_Whole_Genome_Include_Autosomal_Sites_d50.angsdsites -only_proper_pairs 1 -nThreads 8 -out results_saf_subpops/ME_01_Int4 -doSaf 2 -GL 1 -skipTriallelic 1 -rf split_fai_20_angsd/final.assembly.homo.fa_4.interval_list

angsd -bam /n/holylfs/LABS/edwards_lab/ashultz/HFWGReseq/whole-genome-reseq/BamLists_NoRel /NY_90_BamList.txt -anc /n/holylfs/LABS/edwards_lab/ashultz/HFWGReseq/gatk/anc_fasta/HF_ancestral_alleles.fasta -uniqueOnly 1 -remove_bads 1 -trim 0 -minMapQ 20 -minQ 20 -sites /n/holylfs/LABS/edwards_lab/ashultz/HFWGReseq/genome/bedfiles/All_HF_Whole_Genome_Include_Autosomal_Sites_d50.angsdsites -only_proper_pairs 1 -nThreads 8 -out results_saf_subpops/NY_90_Int13 -doSaf 2 -GL 1 -skipTriallelic 1 -rf split_fai_20_angsd/final.assembly.homo.fa_13.interval_list

angsd -bam /n/holylfs/LABS/edwards_lab/ashultz/HFWGReseq/whole-genome-reseq/BamLists_NoRel /NY_90_BamList.txt -anc /n/holylfs/LABS/edwards_lab/ashultz/HFWGReseq/gatk/anc_fasta/HF_ancestral_alleles.fasta -uniqueOnly 1 -remove_bads 1 -trim 0 -minMapQ 20 -minQ 20 -sites /n/holylfs/LABS/edwards_lab/ashultz/HFWGReseq/genome/bedfiles/All_HF_Whole_Genome_Include_Autosomal_Sites_d50.angsdsites -only_proper_pairs 1 -nThreads 8 -out results_saf_subpops/NY_90_Int18 -doSaf 2 -GL 1 -skipTriallelic 1 -rf split_fai_20_angsd/final.assembly.homo.fa_18.interval_list